ruby/ql/lib/codeql/ruby/frameworks/Array.qll

```diff
@@ -0,0 +1,2407 @@
+/** Provides flow summaries for the `Array` and `Enumerable` classes. */
+
+private import codeql.ruby.AST
+private import codeql.ruby.ApiGraphs
+private import codeql.ruby.DataFlow
+private import codeql.ruby.dataflow.FlowSummary
+private import codeql.ruby.dataflow.internal.DataFlowDispatch
+
+private class ArrayIndex extends int {
+  ArrayIndex() { this = any(DataFlow::Content::KnownArrayElementContent c).getIndex() }
+}
+
+/**
+ * Provides flow summaries for the `Array` class.
+ *
+ * The summaries are ordered (and implemented) based on
+ * https://ruby-doc.org/core-3.1.0/Array.html, however for methods that have the
```

Comment:
I think we should probably stop using `ruby-doc.org` as a source of truth. We've encountered several spurious entries already, and I don't think it's an official project.

https://docs.ruby-lang.org/en/3.1/Array.html is probably a better source.